lib/config_cat/config.ex

```diff
@@ -2,19 +2,27 @@ defmodule ConfigCat.Config do
   @moduledoc """
   Defines configuration-related types used in the rest of the library.
   """
-  alias ConfigCat.RedirectMode
+  alias ConfigCat.Config.Preferences
+  alias ConfigCat.Config.Segment
+  alias ConfigCat.Config.Setting
 
   @typedoc false
   @type comparator :: non_neg_integer()
 
   @typedoc "The name of a configuration setting."
   @type key :: String.t()
 
-  @typedoc "The configuration settings within a Config."
-  @type settings :: map()
+  @typedoc false
+  @type opt :: {:preferences, Preferences.t()} | {:settings, settings()}
+
+  @typedoc false
+  @type salt :: String.t()
+
+  @typedoc false
+  @type settings :: %{String.t() => Setting.t()}
 
-  @typedoc "A collection of configuration settings and preferences."
-  @type t :: map()
+  @typedoc "A collection of feature flags and preferences."
```

Comment:
```suggestion
  @typedoc "A collection of configuration settings and preferences."
```
I would keep the original here